src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -1505,12 +1636,179 @@ namespace Microsoft.FSharp.Core
                     override iec.Equals(x:obj,y:obj) = GenericEqualityObj true iec (x,y)  // ER Semantics
                     override iec.GetHashCode(x:obj) = raise (InvalidOperationException (SR.GetString(SR.notUsedForHashing))) }
 
+            let canUseDefaultEqualityComparer (ty:Type) =
+                let processed = System.Collections.Generic.HashSet ()
+
+                let rec recurse idx (types:array<Type>) =
```

Comment:
Please use `Type[]`

Reply:
Really? I haven't been following any F# formatting recommendation discussion but find that a bit surprising. I really like the consistency of `array` with `list` and just generics in general. Anyway, if that's the way that it goes then happy enough to change it.